sqlglot/optimizer/scope.py

```diff
@@ -353,10 +357,27 @@ def selects(self):
         Returns:
             list[exp.Expression]: expressions
         """
-        if isinstance(self.expression, exp.Union):
-            return self.expression.unnest().selects
+        if isinstance(self.expression, exp.Table):
+            return []
         return self.expression.selects
 
+    @property
+    def named_selects(self):
+        """
+        Named select expressions of this scope.
+
+        For example, for the following expression:
+            SELECT 1 as a, 2 as b FROM x
+
+        The outputs are the "a" and " b" expressions.
+
+        Returns:
+            list[exp.Expression]: expressions
+        """
+        if isinstance(self.expression, exp.Table):
```

Comment:
just add named_selects to table directly

Reply:
should we get rid of both `scope.selects` and `scope.named_selects` then for consistency? we can just do `scope.expression.[named_]selects` in places where we need it

Reply:
yea

Reply:
we can get rid

Reply:
got rid of 'em both